content/learn/book/modular-architecture/plugins.md

```diff
@@ -5,10 +5,122 @@ insert_anchor_links = "right"
 weight = 0
 +++
 
-WIP
+**Plugins** are the most important tool for organizing your Bevy projects,
+allowing you to split logic across files, easily toggle behavior for testing,
+and reuse functionality between projects.
 
-When a plugin is added though `App::add_plugins`, the app calls `Plugin::build`, and the plugin typically accesses and configures the world.  Then, when the app is run, a few other plugin life-cycle functions are called, and finally we enter the run loop:
-+ The app polls `Plugin::finished` until all the added plugins return `true`.
-+ The app calls `Plugin::finish` on all plugins.
-+ The app calls `Plugin::cleanup` an all plugins. 
-+ The app calls the run loop function on itself.
+At its heart, the `Plugin` trait is extremely simple.
+It looks a little something like this:
+
+```rust
+trait Plugin {
+    fn build(&self, app: &mut App) {}
+}
+```
+
+Let's break that down:
+
+- this is a trait, so we need to implement it for a user-defined type
+- `build` takes `&self`, allowing you to change behavior based on the value of our type
+- it takes a`&mut App` reference, allowing you to mutate the [`App`] state freely
+  - adding systems via [`App::add_systems`] is the most common and important method
```

Comment:
Maybe worth mentioning observers as well? I'd say they are equally as important in game code (not library code)

Reply:
I'd like to avoid pulling in new concepts here if possible.